lib/library.rb

```diff
@@ -0,0 +1,57 @@
+require './lib/book'
+require './lib/author'
+
+class Library
+
+    attr_reader :name, :books, :authors
+
+    def initialize(name, books = [], authors = [])
+        @name = name
+        @books = books
+        @authors = authors
+    end
+
+    def add_author(author)
+        @authors << author
+        author.books.each do |book|
```

Comment:
What happens if the author writes a book after they are added to the library though? How would that book get added? Instead of making @books, we can just make a method called books that calculates the books dynamically based on @authors. The less instance variables, the better.